### ISSUE TITLE
[Ruby] Lowering memory consumption when writing large file

### ISSUE THREAD

**Issue by cliffano** [swagger-api/swagger-codegen#5704]
##### Description

Currently the generated Ruby client buffers the response body before writing the file in one go.
I tested this by using the Ruby client to download a file where response body's size is ~2Gb, and I monitored the ruby process spiked to ~2Gb memory consumption. This obviously won't scale well with larger files, and is a problem when the machine doesn't have large memory capacity, plus there's going to be a limit of how large a string can be, and how much memory the ruby process can use.

Is there any particular reason why the whole response body for a file type has to be received as a whole, deserialised, and then written to a file in one go?

If the approach further below in 'Suggests a Fix' section looks reasonable, I'm happy to take a stab at it.

##### Swagger-codegen version

master branch.

##### Swagger declaration file content or url

Sample spec segment:
```yaml
  /some/path/to/endpoint:
    get:
      operationId: getFile
      produces:
        - application/octet-stream
      parameters:
       responses:
        default:
          schema:
            type: file
```

##### Steps to reproduce

1. Create an endpoint that returns a large file as part of its response body.
2. Create spec with response schema `type: File` .
3. Generate Ruby client.
4. Call the API and monitor the process' memory usage.

##### Related issues

Couldn't find anything obvious.

##### Suggest a Fix

When Typhoeus::Request is created https://github.com/swagger-api/swagger-codegen/blob/35d3fb82ce82e86370ca24c21b0c066e8aea2f90/modules/swagger-codegen/src/main/resources/ruby/api_client.mustache#L120 , it should be passed `on_headers`, `on_body`, and `on_complete` callbacks if `return_type` is `File`.

**Comment by wing328** [comment]
@cliffano I agree with you the current approach used by the Ruby API client is not ideal so please file a PR with the suggested fix.

**Comment by cliffano** [comment]
@wing328 Raise PR https://github.com/swagger-api/swagger-codegen/pull/5718 .

**Comment by wing328** [comment]
Closed via #5718
